src/content/Blog/running-vllm-on-akash/index.md

```diff
@@ -0,0 +1,143 @@
+
+pubDate: "2024-5-8"
+draft: true
+
+categories:
+  - Insights
+  - Updates
+  - News
+tags:
+  - Insights
+  - Updates
+  - News
+
+contributors:
+  - Logan Cerkovnik
+  - Anil Murty
+
+bannerImage: ./banner-image.jpg
+---
+# Running vLLM on Akash
+By Logan Cerkovnik and Anil Murty
+
+There has been a proliferation of LLM services over the last several months and it’s great to see some of these be made available open source. Ollama is one of the early solutions that gained a significant amount of popularity among developers and has helped many developers accelerate their AI application development using open source AI models. A more recent solution is vLLM - that aims to overcome some of the limitations of Ollama. This post delves into what vLLM is and when and why developers should consider using it. Lastly it also demonstrates how you can run vLLM easily on Akash Network
+
+## Delving into vLLM
+vLLM is an LLM server implementation first introduced in a paper last year. Its primary objective was to make LLM inference faster for multiuser services. What vLLM does also achieve is overcoming two of the limitations of Ollama. vLLM enables you to serve more than 1 user at a time, natively, without having to proxy user requests between multiple GPUs. It also allows easy switching of models back and forth, without blocking users from accessing their specific model.
+
+The main change vLLM makes is adding Paged Attention to LMM model by swapping out all the transformer attention modules for Paged Attention which implements attention more efficiently. The authors of vLLm describe Page Attention as, “PageAttention’s memory sharing greatly reduces the memory overhead of complex sampling algorithms, such as parallel sampling and beam search, cutting their memory usage by up to 55%. This can translate into up to 2.2x improvement in throughput”.  You can read more about the technical details of paged attention on the vllm blog at https://blog.vllm.ai/2023/06/20/vllm.html.  Today, the current server implementation has gone beyond just Paged Attention and will soon support speculative encoding approaches. Other open source alternatives to vLLM include HuggingFace’s TGI and  the sglang engine with its Radix Attention implementation.  The only drawback to using vLLM is that it doesn’t support all of the super low quantization methods and file formats such as GGUF.  If you haven’t used GGUF before in llama.cpp-based tools like ollama then you should note that most people actively try to avoid using models with quantization lower than the 4bit (Q4) quantization due to performance issues.  The good news is that most models are available in GPTQ or AWQ quantization formats that are supported by vLLM. 
```

Comment:
```suggestion
The main change vLLM makes is adding Paged Attention to LLM model by swapping out all the transformer attention modules for Paged Attention which implements attention more efficiently. The authors of vLLM describe Page Attention as, “PageAttention’s memory sharing greatly reduces the memory overhead of complex sampling algorithms, such as parallel sampling and beam search, cutting their memory usage by up to 55%. This can translate into up to 2.2x improvement in throughput”.  You can read more about the technical details of paged attention on the vllm blog at https://blog.vllm.ai/2023/06/20/vllm.html.  Today, the current server implementation has gone beyond just Paged Attention and will soon support speculative encoding approaches. Other open source alternatives to vLLM include HuggingFace’s TGI and the sglang engine with its Radix Attention implementation.  The only drawback to using vLLM is that it doesn’t support all of the super low quantization methods and file formats such as GGUF.  If you haven’t used GGUF before in llama.cpp-based tools like ollama then you should note that most people actively try to avoid using models with quantization lower than the 4bit (Q4) quantization due to performance issues.  The good news is that most models are available in GPTQ or AWQ quantization formats that are supported by vLLM. 
```

Reply:
Changed
- LMM to LLM
- vLLm to vLLM

Reply:
I ave fixed this